homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,165 @@
+
+    # Add a subscriber to the coordinator to discover new devices
+    coordinator.async_add_listener(async_update_data)
+
+
+class IottyLightSwitch(SwitchEntity, CoordinatorEntity[Device]):
+    """Haas entity class for iotty LightSwitch."""
+
+    _attr_has_entity_name = True
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _iotty_cloud: IottyProxy
+    _iotty_device: LightSwitch
+
+    def __init__(
+        self,
+        coordinator: IottyDataUpdateCoordinator,
+        iotty_cloud: IottyProxy,
+        iotty_device: LightSwitch,
+    ) -> None:
+        """Initialize the LightSwitch device."""
+        super().__init__(coordinator=coordinator)
+
+        _LOGGER.debug(
+            "Creating new SWITCH (%s) %s",
+            iotty_device.device_type,
+            iotty_device.device_id,
+        )
+
+        self._iotty_cloud = iotty_cloud
+        self._iotty_device = iotty_device
+        self._attr_unique_id = iotty_device.device_id
+
+    @property
+    def device_id(self) -> str:
+        """Get the ID of this iotty Device."""
+        return self._iotty_device.device_id
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the LightSwitch is on."""
+        _LOGGER.debug(
+            "Retrieve device status for %s ? %s",
+            self._iotty_device.device_id,
+            self._iotty_device.is_on,
+        )
+        return self._iotty_device.is_on
+
+    @property
+    def name(self) -> str:
+        """Get the name of this iotty Device."""
+        return self._iotty_device.name
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the LightSwitch on."""
+        _LOGGER.debug("[%s] Turning on", self._iotty_device.device_id)
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id, self._iotty_device.cmd_turn_on()
+        )
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the LightSwitch off."""
+        _LOGGER.debug("[%s] Turning off", self._iotty_device.device_id)
+        await self._iotty_cloud.command(
+            self._iotty_device.device_id, self._iotty_device.cmd_turn_off()
+        )
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        try:
+            device: LightSwitch = next(
+                device
+                for device in self.coordinator.data.devices
+                if device.device_id == self._iotty_device.device_id
+            )
+            self._iotty_device.is_on = device.is_on
+            self.async_write_ha_state()
+        except StopIteration:
+            _LOGGER.debug(
+                "Device with id [%s] is no longer handled by iotty, please remove it from hass",
+                self._iotty_device.device_id,
+            )
```

Comment:
Who are we asking to remove the device?

Reply:
This is just a console error that the user might see if they decide to change the log level for the integration. Would you prefer a different wording?

Reply:
Since the integration now cleans up, deletion this is automatically done, correct? I'd suggest to just remove the `,please remove it from hass" part of the message.